### PR TITLE
Fix 867 by rejecting unsupported OpenAPI specs

### PR DESCRIPTION
**Pull request by mromaszewicz**

### CODE REVIEW

**Review by jamietanna — COMMENTED** [review]
Is this not going to be a breaking change? Given we've had to update our test files

Does this impact cases we've got partial specs (ie for external refs)?

I'd seen that when I started with https://github.com/oapi-codegen/oapi-codegen/pull/713

**Comment by mromaszewicz** [comment]
> Is this not going to be a breaking change? Given we've had to update our test files
> 
> Does this impact cases we've got partial specs (ie for external refs)?
> 
> I'd seen that when I started with #713

Yes. It is indeed a breaking change, and one I wanted your opinion on.

A more compatible way would be to shim ourselves into the loading path in Kin somehow, and only complain if there's a known incompatible version present, and accept specs without versions. Let me have a second go at it...

File: cmd/oapi-codegen/oapi-codegen.go [hunk 1]
```diff
@@ -281,20 +282,41 @@ func main() {
 		return
 	}
 
-	swagger, err := util.LoadSwagger(flag.Arg(0))
+	spec, err := util.LoadOpenAPI(flag.Arg(0))
 	if err != nil {
 		errExit("error loading swagger spec in %s\n: %s\n", flag.Arg(0), err)
```
Comment (ccoVeille) [inline cmd/oapi-codegen/oapi-codegen.go:287]: Shouldn't be this?

```suggestion
 		errExit("error loading OpenAPI spec in %s\n: %s\n", flag.Arg(0), err)
```

Comment (jamietanna) [reply cmd/oapi-codegen/oapi-codegen.go:287]: It's unfortunately a wider issue https://github.com/oapi-codegen/oapi-codegen/issues/672

**Comment by mromaszewicz** [comment]
I'm going to do this differently, without affecting the loading of broken specs.

**Pull request closed without merging**